### DISCUSSION TITLE
Enabling JPEG XL with Vips on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by dkam** [libvips/libvips#3700] [Q&A]
Hi there,
    I'm trying to get RubyVIPS to support JPEG XL on Ubuntu 22.04.  Ubuntu doesn't have libjxl in its repository, so I've installed the packages from their [releases](https://github.com/libjxl/libjxl/releases/) page. 
    
`sudo dpkg -i libjxl_0.8.2_amd64.deb libjxl-dev_0.8.2_amd64.deb  jxl_0.8.2_amd64.deb`
   
The `cjxl` binary is available. 

pkg-config seems to be able to find it
```shell
$ pkg-config --libs --cflags libjxl
-DHWY_SHARED_DEFINE -ljxl
```

Vips still can't see the jxl library though. 
```shell
$ vips -l | grep jxl
$
```
Any tips on how to get libvips to support jxl on Ubuntu 22.04?

Once I've got that working, I'd like to get ruby-vips to support jxl - it currently returns `VipsOperation: class "jxlsave" not found (Vips::Error)`

**Comment by dkam** [comment]
Experimenting with building libvips on Ubuntu 22.04, with a prefix of ~/libvips, after running  meson install, I see a ~/libvips/bin/vips executable, but it references the system vips - version `vips-8.12.1`.  I can see the build-dir/toolsvips is `vips-8.14.5`

The `build-dir/tools/vips` does support jxl - is there a bug with the install?  Why is the installed `vips` executable the version which ships with Ubuntu 22.04?

**Comment by dkam** [comment]
On the plus side, setting `LD_LIBRARY_PATH=/home/deploy/libvips/lib/x86_64-linux-gnu` lets RubyVips use the built version of Vips which supports JXL 🎉

**Comment by dkam** [comment]
I guess I can almost close this Q&A.  The packaged version of vips doesn't dynamically load libjxl.  The last question is - why did the `meson install` install the wrong version of the `vips` binary into the prefix path?

**Reply by jcupitt** [reply]
Hi @dkam,

As you found, you need to set `PATH` *and* `LD_LIBRARY_PATH`. Executables will get libraries from the system areas if you don't.

Yes, 22.04 does not have libjxl, unfortunately, but they did start bundling it with 23.04.
